src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -243,6 +245,16 @@ public static void parseField(AbstractFieldMapper.Builder builder, String name,
                 parseCopyFields(propNode, builder);
             }
         }
+        if (indexAnalyzer != null) {
+            builder.indexAnalyzer(indexAnalyzer);
+        } else {
+            builder.indexAnalyzer(theAnalyzer);
+        }
+        if (searchAnalyzer != null) {
+            builder.searchAnalyzer(searchAnalyzer);
+        } else {
+            builder.searchAnalyzer(theAnalyzer);
```

Comment:
You could make it the same with an `else if` instead of `else`:
```
} else if (theAnalyzer != null) {
  builder.searchAnalyzer(theAnalyzer);
}
```